File: dp_primer/dp_primer_stairs_step0.cs
// 実行時間 40ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int n = int.Parse( ReadLine() );
        int[] a = new int[41];

        a[0] = 1;
        foreach( int i in Range( 1, n ) ) {
            a[i] = 0;
            if( 1 <= i ) {
                a[i] += a[i - 1];
            }
            if( 2 <= i ) {
                a[i] += a[i - 2];
            }
        }
        WriteLine( a[n] );
        return;
    }
}

File: dp_primer/dp_primer_stairs_step0.py
# 実行時間 20ms
n = int(input())
dp = [1]

for i in range(n + 1):
    dp.append(0)
    if 1 <= i:
        dp[i] += dp[i - 1]
    if 2 <= i:
        dp[i] += dp[i - 2]
print(dp[n])
